### PR TITLE
Ops: Added shrink_volume op

### PR DESCRIPTION
**Pull request by nik-redhat**
### All Submissions:

Description:
Added the shrink_volume and form_brick_list_to_remove_bricks op

Fixes: #736

Signed-off-by: nik-redhat <nladha@redhat.com>

### CODE REVIEW

**Comment by nik-redhat** [comment]
Test run:
```
Table:
 test_expand_shrink_volume
 
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time taken (HH:MM:SS) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     dist    |     PASS    |        00:00:47       |     N/A     |
|     rep     |     PASS    |        00:01:08       |     N/A     |
|   dist-rep  |     PASS    |        00:01:15       |     N/A     |
|     arb     |     PASS    |        00:01:09       |     N/A     |
|   dist-arb  |     PASS    |        00:01:15       |     N/A     |
|     disp    |     PASS    |        00:02:07       |     N/A     |
|  dist-disp  |     PASS    |        00:02:23       |     N/A     |
+-------------+-------------+-----------------------+-------------+



Framework runtime : 00 hours 10 minutes 18 seconds
```

File: common/ops/gluster_ops/volume_ops.py [hunk 1]
```diff
@@ -477,6 +477,192 @@ def expand_volume(self, node: str, volname: str, server_list: list,
...
+                         f"{bricks_list_to_remove} on volume {volname}")
+
+        # Delete the removed bricks
+        if delete_bricks:
+            ret = self.delete_bricks(bricks_list_to_remove)
+            if not ret:
+                return False
+            for brick in bricks_list_to_remove:
+                node, brick_dir = brick.split(':')
+                self.es.remove_val_from_cleands(node, brick_dir)
```
Comment (srijan-sivakumar) [inline common/ops/gluster_ops/volume_ops.py:662]: We might also need to handle the changes to the volume counts, i.e. rep_count, dist_count etc in environ..

**Review by srijan-sivakumar — CHANGES_REQUESTED** [review]

File: common/ops/gluster_ops/volume_ops.py (hunk 1 above)
Comment (nik-redhat) [reply common/ops/gluster_ops/volume_ops.py:662]: Yeah, will add that.

**Comment by nik-redhat** [comment]
Test run:
```
Table:
 test_expand_shrink_volume
 
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time taken (HH:MM:SS) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:01:01       |     N/A     |
|     dist    |     PASS    |        00:00:54       |     N/A     |
|   dist-rep  |     PASS    |        00:01:16       |     N/A     |
|     arb     |     PASS    |        00:01:09       |     N/A     |
|   dist-arb  |     PASS    |        00:01:15       |     N/A     |
|     disp    |     PASS    |        00:02:10       |     N/A     |
|  dist-disp  |     PASS    |        00:02:20       |     N/A     |
+-------------+-------------+-----------------------+-------------+



Framework runtime : 00 hours 10 minutes 19 seconds
```

**Comment by nik-redhat** [comment]
I think we are good to take in this PR as well?

**Review by aujjwal-redhat — APPROVED** [review]
LGTM

**Review by srijan-sivakumar — APPROVED** [review]

**Pull request merged**